ipa-core/src/protocol/ipa_prf/shuffle/sharded.rs

```diff
@@ -234,12 +198,16 @@ impl<C: ShardedContext> ShuffleContext for C {}
 ///
 /// [`ShuffleShare`] and [`Shuffleable`] are added to bridge the gap. They can be implemented for
 /// arbitrary structs as long as `Add` operation can be defined on them.
-pub trait ShuffleShare: Sendable + Clone + FromRandom + Add<Output = Self> {}
+pub trait ShuffleShare: BooleanArray + Serializable + FromRandom {}
```

Comment:
I think `BooleanArray` bound may be problematic for structs like `HybridReport`, would it work if we ask implementors to give us bit representation of their data? something like `fn as_bytes_iter(&self) -> impl Iterator<Item = [u8; 32]>`. iirc we only need it to compute MACs

Reply:
Note that these are the trait bounds for `ShuffleShare`, not for `Shuffleable` itself (i.e. it is saying that any `Shuffleable` type needs to support conversion to boolean arrays, not that only boolean arrays are shuffleable). That said, after some back-and-forth with myself in #1395, I was able to arrive at a more precise set of bounds for `ShuffleShare`, which you can see in #1397 at line 209 of sharded.rs.

#1397 also has the implementations of `Shuffleable` for `AttributionOutputs` and `OPRFIPAInputRow`.